### PR TITLE
Migrate /static/alert.js to TypeScript

### PR DESCRIPTION
**Pull request by supergrecko**
Migrates the alert widget to TypeScript

Non-functional change.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,6 @@
     "@sentry/browser": "^6.4.1",
     "@sentry/node": "^6.4.1",
     "@sentry/tracing": "^6.4.1",
-    "@types/underscore": "^1.11.2",
```
Comment (supergrecko) [inline package.json:24]: Moved to dev dependencies

**Comment by partouf** [comment]
Oof, I'm totally here for this revamp, but I also have changes in this file. :'(
I guess I'll have to suffer through the merge conflicts...

Also if you're gonna make interfaces, I would love for them to be in a separate file. `alert.interfaces.ts` or something

**Comment by supergrecko** [comment]
> Oof, I'm totally here for this revamp, but I also have changes in this file. :'(
> I guess I'll have to suffer through the merge conflicts...
> 
> Also if you're gonna make interfaces, I would love for them to be in a separate file. `alert.interfaces.ts` or something

Unless you plan on adding more stuff to alerts I could happily migrate your `Alert.prototype.enterSomething` in #2725 and include that here.

I'll move the interfaces.

**Comment by partouf** [comment]
> Unless you plan on adding more stuff to alerts I could happily migrate your `Alert.prototype.enterSomething` in #2725 and include that here.

No plans, this should be it. I would be very grateful!

File: static/alert.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import $ from 'jquery';
+
+import { AlertAskOptions, AlertNotifyOptions } from "./alert.interfaces";
+
+export class Alert {
+    yesHandler: (answer?: string | string[] | number) => void | null = null;
```
Comment (supergrecko) [inline static/alert.ts:30]: This `string | string[] | number` type is a little awkward, but that is what JQuery's .val() returns. One option would be to cast to string and type as `string`, but I imagine it's better to let the user decide that themselves.

Comment (mattgodbolt) [reply static/alert.ts:30]: Very weak opinion here as I'm extremely new to TypeScript, but I agree with you here :)

File: views/popups.pug [hunk 3]
```diff
@@ -65,6 +65,21 @@
         button.btn.btn-link.no(type="button" data-dismiss="modal") No
         button.btn.btn-link.yes(type="button" data-dismiss="modal") Yes
 
+#enter-something.modal.fade.gl_keep(tabindex="-1" role="dialog")
```
Comment (mattgodbolt) [inline views/popups.pug:68]: Is this a new thing? From the discussion with @partouf is this you picking up Partouf's separate change and rolling it in? It's :100: ok I'm just trying to understand.

**Review by mattgodbolt — APPROVED** [review]
I'm so very excited to move away from ES5 :)

File: views/popups.pug (hunk 3 above)
Comment (supergrecko) [reply views/popups.pug:68]: Yeah, to reduce the amount of merges @partouf has to suffer through with #2725 we decided to move in his alert.js changes here

Comment (partouf) [reply views/popups.pug:68]: Yes, I made changes in alert.js for the multifileui to rename files, which includes a new modal dialog. @supergrecko was nice enough to implement them here so I won't be scratching my head on how to merge a deleted js files contents etc.

**Pull request merged**